aspnet/security/accconfirm.rst

```diff
@@ -1,236 +1,209 @@
+In **SQL Server Object Explorer** (SSOX), navigate to **(localdb)\MSSQLLocalDB(SQL Server 12)**. Right click on **dbo.AspNetUsers** > **View Data**:
 
+.. image:: accconfirm/_static/ssox.png
+.. image:: accconfirm/_static/au.png
 
-Running the Application
------------------------
+Note the ``EmailConfirmed`` field is ``False``.
 
-Run the application and after a quick build step, you should see it open in your web browser.
+Right click on the row and select delete. You'll might want to use this email again in the next step, when the app sends a confirmation email.
 
-.. image:: accconfirm/_static/first-run.png
+Require SSL
+------------------------
 
+In this section we'll set up our Visual Studio project to use SSL and our project to require SSL.
+
+Enable SSL in Visual Studio
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. From the **Project** menu, select **Web app Properties**. 
+2. Select **Debug** in the left pane (see image below).
+3. Check **Enable SSL**, and then save changes (this step is necessary to populate the URL box).
+4. Copy the **URL** and paste it into the **Launch URL** box.
+
+.. image:: accconfirm/_static/ssl.png
+
+Require HTTPS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add the ``[RequireHttps]`` attribute to each controller. The ``[RequireHttps]`` attribute will redirect all HTTP GET requests to HTTPS GET and will reject all HTTP POSTs. A security best practice is to use HTTPS for all requests.
+
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Controllers/HomeController.cs
+   :language: c#
+   :lines: 9-10
+   :emphasize-lines: 9
+   :dedent: 4
+
+Require email confirmation
+----------------------------
+
+It's a best practice to confirm the email of a new user registration to verify they are not impersonating someone else (that is, they haven't registered with someone else's email). Suppose you had a discussion forum, you would want to prevent "bob@example.com" from registering as "joe@contoso.com". Without email confirmation, "joe@contoso.com" could get unwanted email from your app. Suppose Bob accidentally registered as  "bib@example.com" and hadn't noticed it, he wouldn't be able to use password recover because the app doesn't have his correct email. Email confirmation provides only limited protection from bots and doesn't provide protection from determined spammers, they have many working email aliases they can use to register.
+
+You generally want to prevent new users from posting any data to your web site before they have been confirmed by email, a SMS text message or another mechanism. In the sections below, we will enable email confirmation and modify the code to prevent newly registered  users from logging in until their email has been confirmed.
```

Comment:
Missing serial (Oxford) comma after "SMS text message":
You generally want to prevent new users from posting any data to your web site before they have been confirmed by email, a SMS text message or another mechanism.